apps/web/hooks/useBucket.tsx

```diff
@@ -25,22 +25,19 @@ export const ROOT_FOLDER: DriveFolder = {
 };
 
 export default function useBucket(): ContextValue {
-  const { keys } = useKeys();
-
-  if ((Provider[keys.type] as Provider) === Provider.firebase) {
-    return useFirebase();
-  } else if ((Provider[keys.type] as Provider) === Provider.s3) {
-    return useS3();
-  } else if ((Provider[keys.type] as Provider) === Provider.backblaze) {
-    return useS3();
-  } else if ((Provider[keys.type] as Provider) === Provider.cloudflare) {
-    return useS3();
-  } else if ((Provider[keys.type] as Provider) === Provider.wasabi) {
-    return useS3();
-  } else if ((Provider[keys.type] as Provider) === Provider.digitalocean) {
-    return useS3();
-  }
-
-  return null;
+	const { keys } = useKeys();
+  
+	switch (Provider[keys.type] as Provider) {
+	  case Provider.firebase:
+		return useFirebase();
+	  case Provider.s3:
+	  case Provider.backblaze:
+	  case Provider.wasabi:
+	  case Provider.digitalocean:
+    case Provider.cloudflare:
```

Comment:
why do we have CloudFlare in this PR? can we focus on digital ocean and wasabi before merging this?

Reply:
This was the owner's changes that were merged into the test branch

Reply:
so this PR is a mix of you/nathan's changes and the owners? can we specifically merge the owner's changes into the repo (with or without a PR) and then open one for your changes?

According to [this article](https://fagnerbrack.com/one-pull-request-one-concern-e84a27dfe9f1#:~:text=While%20a%20commit%20can%20only,form%20a%20high%2Dlevel%20concern.&text=It%20is%20also%20important%20for%20a%20Pull%20Request%20to%20be%20atomic.), "A Pull Request represents a way to deliver value to the application in the form of a set of changes that together form a high-level concern."

if we are able to apply a separation of concerns to pull requests, that makes these changes make sense.

Reply:
@9brian FYI: if you need to re-request me for a review, there is a button next to the reviewer's name that allows you to do so

Reply:
like you're allowed to disagree with me but robust descriptions help engineers understand your POV